### PR TITLE
Add azure-pipelines-release.yml (main)

### PR DESCRIPTION
**Pull request by bachuv**
Adding azure-pipelines-release.yml to main

### Pull request checklist

* [x] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation PR is ready to merge and referenced in `pending_docs.md`
* [x] My changes **should not** be added to the release notes for the next release
    * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)
* [x] My changes **do not** require any extra work to be leveraged by OutOfProc SDKs
    * [ ] Otherwise: That work is being tracked here: #issue_or_pr_in_each_sdk

### CODE REVIEW

File: azure-pipelines-release.yml
Comment (amdeel) [inline azure-pipelines-release.yml:27]: Why did we add a solution file here? We should just use the existing solution file.

**Comment by cgillum** [comment]
I noticed you removed the .NET 6 tools in bd60749. Is it still installed on the VM? We're going to need .NET 6 after [this PR](https://github.com/Azure/azure-functions-durable-extension/pull/2046) is merged.

**Comment by davidmrdavid** [comment]
@cgillum, .NET 6 is already included in the VM's dependencies, so there's a good chance that your new PR will just be able to pick it up. If it doesn't, we can add the `UseDotNet` task for version 6 to the `.yml` deffinition.

**Review by cgillum — APPROVED** [review]
Looks good to me. Let's get @amdeel's sign-off as well before merging.

**Review by amdeel — APPROVED** [review]
LGTM!

**Pull request merged**
